code/modules/reagents/chemistry/reagents/drugs.dm

```diff
@@ -801,6 +801,302 @@
+					var/obj/item/organ/internal/our_heart = datum_heart.linked_organ
+					our_heart.receive_damage(0.15 * constant_dose_time, TRUE) // Basically you might die. Especially if you are a slime.
+				else
+					handle_heartless(L, 0.15 * constant_dose_time)
+
+
+	if(!L.hud_used)
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = L.hud_used?.plane_master_controllers[PLANE_MASTERS_GAME]
+
+	game_plane_master_controller.remove_filter(MEPHEDRONE_SCREEN_FILTER)
+	game_plane_master_controller.remove_filter(MEPHEDRONE_SCREEN_BLUR)
+
+
+/// Leaves an afterimage behind the mob when they move
+/datum/reagent/twitch/proc/on_movement(mob/living/carbon/L, atom/old_loc)
+	SIGNAL_HANDLER
+	if(HAS_TRAIT(L, TRAIT_IMMOBILIZED)) //No, dead people floating through space do not need afterimages
+		return NONE
+	var/overdosed = (id in L.reagents.overdose_list())
+	new /obj/effect/temp_visual/decoy/twitch_afterimage(old_loc, L, overdosed? 1.25 SECONDS : 0.75 SECONDS)
+
+/// Tries to dodge incoming bullets if we aren't disabled for any reasons
+/datum/reagent/twitch/proc/dodge_bullets(mob/living/carbon/human/source, obj/item/projectile/hitting_projectile)
+	SIGNAL_HANDLER
+
+	if(HAS_TRAIT(source, TRAIT_IMMOBILIZED))
+		return NONE
+	source.visible_message(
+		"<span class='danger'>[source] effortlessly dodges [hitting_projectile]!</span>",
+		"<span class='userdanger'>You effortlessly evade [hitting_projectile]!</span>",
+	)
+	playsound(source, pick('sound/weapons/bulletflyby.ogg', 'sound/weapons/bulletflyby2.ogg', 'sound/weapons/bulletflyby3.ogg'), 75, TRUE)
+	source.add_filter(MEPHEDRONE_BLUR_EFFECT, 2, gauss_blur_filter(5))
+	addtimer(CALLBACK(source, TYPE_PROC_REF(/atom, remove_filter), MEPHEDRONE_BLUR_EFFECT), 0.5 SECONDS)
+	return ATOM_PREHIT_FALSE
+
+/datum/reagent/twitch/proc/no_hud_cheese(mob/living/carbon/L)
+	SIGNAL_HANDLER
+	addtimer(CALLBACK(src, PROC_REF(no_hud_cheese_2), L), 2 SECONDS) //Calling it instantly will not work, need to give it a moment
+
+
+/datum/reagent/twitch/proc/no_hud_cheese_2(mob/living/carbon/L) //Basically if you change the UI you would remove the visuals. This fixes that.
+	var/atom/movable/plane_master_controller/game_plane_master_controller = L.hud_used?.plane_master_controllers[PLANE_MASTERS_GAME]
+	game_plane_master_controller.remove_filter(MEPHEDRONE_SCREEN_FILTER)
+	game_plane_master_controller.remove_filter(MEPHEDRONE_SCREEN_BLUR)
+
+	var/static/list/col_filter_green = list(0.5,0,0,0, 0,1,0,0, 0,0,0.5,0, 0,0,0,1)
+
+	game_plane_master_controller.add_filter(MEPHEDRONE_SCREEN_FILTER, 10, color_matrix_filter(col_filter_green, FILTER_COLOR_RGB))
+
+	game_plane_master_controller.add_filter(MEPHEDRONE_SCREEN_BLUR, 1, list("type" = "radial_blur", "size" = 0.1))
+
+	var/overdosed = (id in L.reagents.overdose_list())
+	if(overdosed)
+		addtimer(CALLBACK(src, PROC_REF(no_hud_cheese_3), L), 1 SECONDS) //still needs a moment
+
+/datum/reagent/twitch/proc/no_hud_cheese_3(mob/living/carbon/L)
```

Comment:
Can you maybe document what some of these hud cheeses will do?

Reply:
Sure!

Reply:
done